:books: docs documentation overhaul

JPchico opened the pull request:
(WIP) Improvements to the documentation before the release.

JPchico commented:
Hi @chrisjsewell  and @sphuber I have finally had some time to work on the documentation. It is a work in progress and I plan to be making more changes and additions in the coming days.

I drew quite heavily from the QE plugin structure and formatting as I think it is the most complete right now. 

Let me know if you think that the structure makes sense and I'll keep updating regularly.

chrisjsewell commented:
> Could you try addressing the failure.

Yeh looks like you need to fix some of the `toctree` 

```
pickling environment... /home/docs/checkouts/readthedocs.org/user_builds/aiida-lammps/checkouts/86/docs/source/index.md:25: WARNING: toctree contains reference to nonexisting document 'reference/api/index'
done
checking consistency... /home/docs/checkouts/readthedocs.org/user_builds/aiida-lammps/checkouts/86/docs/source/developers/intro.md: WARNING: document isn't included in any toctree
/home/docs/checkouts/readthedocs.org/user_builds/aiida-lammps/checkouts/86/docs/source/reference/api/auto/aiida_lammps/calculations/index.rst: WARNING: document isn't included in any toctree
/home/docs/checkouts/readthedocs.org/user_builds/aiida-lammps/checkouts/86/docs/source/reference/api/auto/aiida_lammps/data/index.rst: WARNING: document isn't included in any toctree
/home/docs/checkouts/readthedocs.org/user_builds/aiida-lammps/checkouts/86/docs/source/reference/api/auto/aiida_lammps/parsers/index.rst: WARNING: document isn't included in any toctree
/home/docs/checkouts/readthedocs.org/user_builds/aiida-lammps/checkouts/86/docs/source/reference/api/auto/aiida_lammps/utils/index.rst: WARNING: document isn't included in any toctree
/home/docs/checkouts/readthedocs.org/user_builds/aiida-lammps/checkouts/86/docs/source/reference/api/auto/aiida_lammps/validation/index.rst: WARNING: document isn't included in any toctree
/home/docs/checkouts/readthedocs.org/user_builds/aiida-lammps/checkouts/86/docs/source/reference/api/auto/aiida_lammps/workflows/index.rst: WARNING: document isn't included in any toctree
/home/docs/checkouts/readthedocs.org/user_builds/aiida-lammps/checkouts/86/docs/source/users/example_md.md: WARNING: document isn't included in any toctree
/home/docs/checkouts/readthedocs.org/user_builds/aiida-lammps/checkouts/86/docs/source/users/example_minimize.md: WARNING: document isn't included in any toctree
/home/docs/checkouts/readthedocs.org/user_builds/aiida-lammps/checkouts/86/docs/source/users/get_started.md: WARNING: document isn't included in any toctree
```

JPchico commented:
Hi! sorry, I did not notice that the building was failing in the CI since it was working locally. I think it should be fixed now.

JPchico commented:
@sphuber and @chrisjsewell I have fixed the issues with the build and I have added a bit more of information. There is still more work needed but I think that it is getting there.

I also changed some of the examples to use the requests library to download the files needed, like the `data.rhodo` for the raw calculation.

sphuber commented on docs/source/_static/logo.pptx:
do we really need a powerpoint file for the logo? 😕

JPchico replied:
I do not even know why that is there :upside_down_face:

sphuber commented on docs/source/index.md:
```diff
@@ -1,37 +1,114 @@
-In this version a major refactoring of the entrie plugin has been done so that render it backwards incompatible with older versions.
+# AiiDA LAMMPS Plugin
+
+{{ aiida_lammps }} is a Python package that allows the workflow management and data provenance tracking framework [AiiDA](http://aiida-core.readthedocs.io/) run {{ LAMMPS }} calculations.
```
```suggestion
{{ aiida_lammps }} is a Python package that allows the workflow management and data provenance tracking framework [AiiDA](http://aiida-core.readthedocs.io/) to run {{ LAMMPS }} calculations.
```

sphuber commented on docs/source/index.md:
```diff
@@ -1,37 +1,114 @@
 ### What does this implies?
 
-Instead of relying on the internal ``LAMMPS`` scripting language to treat loops, multiple phases, definition of custom variables, etc., those tasks are off loaded to the AiiDA worchains, allowing one to make use of the provenance tracking, automated data storage and caching capabilities of AiiDA.
+Instead of relying on the internal {{ LAMMPS }} scripting language to treat loops, multiple phases, definition of custom variables, etc., those tasks are off loaded to the AiiDA workchains, allowing one to make use of the provenance tracking, automated data storage and caching capabilities of AiiDA.
```
Title above should be changed to `What does this imply`

sphuber commented on docs/source/topics/data/potential.md:
```diff
@@ -0,0 +1,77 @@
+# `LammpsPotentialData`
```
```suggestion
# ``LammpsPotentialData``
```

sphuber commented on docs/source/tutorials/first_raw.md:
````diff
@@ -0,0 +1,178 @@
+
+```python
+import requests
+data = SinglefileData(
+    io.StringIO(
+		textwrap.dedent(
+			requests.get(
+				"https://raw.githubusercontent.com/lammps/lammps/develop/bench/data.rhodo"
+			).text
+		)
+    )
+)
````
The indentation was making it really difficult to read and understand
```suggestion
request = requests.get("https://raw.githubusercontent.com/lammps/lammps/develop/bench/data.rhodo")
data = SinglefileData(io.StringIO(request.text)))
```
By the way, in AiiDA v2.5 (to be released soon), we will be able to do `SinglefileData.from_string(request.text)`. No longer the need to import and use `io.StringIO` as the go between. Small usability improvement

JPchico replied:
This is super nice! Looking forward to that feature!

sphuber commented on docs/source/tutorials/include/scripts/run_md_basic.py:
```diff
@@ -0,0 +1,96 @@
+_stream = io.BytesIO(
+    requests.get(
+        "https://openkim.org/files/MO_546673549085_000/Fe_2.eam.fs", timeout=20
+    ).text.encode("ascii")
```
is `ascii` enconding intentional? Shouldn't we just use utf-8, as is the standard across all of AiiDA?

JPchico replied:
I just used ascii as I was not sure, but I think that utf8 should be good.

sphuber commented on docs/source/tutorials/include/scripts/run_raw_basic.py:
```diff
@@ -0,0 +1,74 @@
+        )
+    )
+)
+data = SinglefileData(
+    io.StringIO(
+        textwrap.dedent(
+            requests.get(
+                "https://raw.githubusercontent.com/lammps/lammps/develop/bench/data.rhodo",
+                timeout=20,
+            ).text
+        )
+    )
+)
```
The `textwrap.dedent` doesn't really do anything here if you are not including literal text but a variable
```suggestion
request =  requests.get(
    "https://raw.githubusercontent.com/lammps/lammps/develop/bench/data.rhodo",
    timeout=20,
)
data = SinglefileData(io.StringIO(request.text))
```

sphuber commented on examples/launch_lammps_raw_script.py:
```diff
@@ -47,16 +48,10 @@
-            1034 impropers
-            ...
-            """
+            requests.get(
+                "https://raw.githubusercontent.com/lammps/lammps/develop/bench/data.rhodo",
+                timeout=20,
+            ).text
```
simplify as the others

sphuber reviewed (changes requested):
Thanks a lot @JPchico great work! I have a few minor comments but overall it reads really well and think it is a great improvement. Fine for me to merge after you addresss the few points left in the code

JPchico commented:
@sphuber  and @chrisjsewell  I have made the suggested changes and added a bit more on the workflows and calculations topics. I think that this is a quite good stage for the documentation right now.
If you agree I will merge and close the issue.

After that I will deal with two small things that I realized while dealing with this (will open issues on those now) and I think that then we can make the release. I have  never made a PyPI release, but for what I see it should automatically done when the release is done in the github side. Is that correct?

sphuber commented:
Yep, the [`cd.yml` workflow](https://github.com/aiidaplugins/aiida-lammps/blob/develop/.github/workflows/cd.yml) should take care of that. All you have to do is make a commit that bumps the version number (and update the `CHANGELOG.md`), tag it and then push the tag. The pushed tag will trigger the workflow and if all tests/pre-commit pass, it will build and push to PyPI.

sphuber commented:
I would also propose that after that is done, we simply merge `develop` into `master` and just use `master` as the main development branch and get rid of `develop`. We have been using this approach in all other `aiidateam` and `aiidaplugins` repos and it works perfectly fine. We really don't have the need for a two branch approach.
